### PR TITLE
C++20 bump; Expose event loop and IP/IPrange objects

### PR DESCRIPTION
**Pull request by dr7ana**

### CODE REVIEW

File: include/oxen/quic/ip.hpp [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+    inline constexpr ipv6_net operator/(const ipv6 a, uint8_t mask)
+    {
+        return {a.to_base(mask), mask};
+    }
+
+    inline constexpr ipv4_net ipv4_loopback = ipv4(127, 0, 0, 1) / 8;
+    inline constexpr ipv6 ipv6_loopback(0, 0, 0, 0, 0, 0, 0, 1);
+
+    inline const std::array ipv4_nonpublic = {
```
Comment (jagerman) [inline include/oxen/quic/ip.hpp]: Will this work as `constexpr`?

Comment (dr7ana) [reply include/oxen/quic/ip.hpp]: done

File: tests/001-handshake.cpp [hunk 2]
```diff
@@ -93,29 +93,104 @@ namespace oxen::quic::test
...
-            auto ep = test_net.endpoint(default_addr);
-            // Note: kernel chooses a random port after being passed default addr
-            CHECK_FALSE(ep->local().to_string() == default_addr.to_string());
-        };
+#ifndef _WIN32
+        SECTION("IPv4 Addresses", "[ipv4][constructors][ipaddr]")
+        {
+            uint32_t v4_n;                    // network order ipv4 addr
+            auto v4_h = "192.168.1.1"s;       // host order ipv4 string
+            auto v4_full = "192.168.1.1:0"s;  // full ipv4 addr/port string
```
Comment (tewinget) [inline tests/001-handshake.cpp]: This could be a non-zero port to have symmetry with the ipv6 section after this.

Comment (dr7ana) [reply tests/001-handshake.cpp]: valid, changed :+1:

**Review by tewinget — APPROVED** [review]

**Pull request merged**